src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -40,10 +40,23 @@ public class FhirCosmosSearchService : SearchService
             SearchOptions searchOptions,
             CancellationToken cancellationToken)
         {
-            return await ExecuteSearchAsync(
+            SearchResult searchResult = await ExecuteSearchAsync(
                 _queryBuilder.BuildSqlQuerySpec(searchOptions),
                 searchOptions,
                 cancellationToken);
+
+            if (searchOptions.IncludeTotal == TotalType.Accurate && !searchOptions.CountOnly)
+            {
+                searchOptions.AsCountOnly(true);
+                var totalSearchResult = await ExecuteSearchAsync(
```

Comment:
It seems like what we want is the ability to tell the search implementation that we want numbers not the actual list of results. I think we can refactor the code a little bit so that it's not "forced" to fit the exsiting call patterns.

E.g., maybe keep SearchOptions strickly result of parsing the user input so it reflects what user has input and allow the search to perform in various different ways through other parameters or options. 

Does that make sense?

Reply:
As discussed, I plan to refactor to avoid modifying the SearchOptions object in this file.

Reply:
Implemented in 3357067 and d43999e.